### PR TITLE
Add resource and datasource mso_schema_site_l3out (DCNE-149)

### PR DESCRIPTION
**Pull request by lhercot**
This PR is a part of the phase-2-resources PR

### CODE REVIEW

File: website/docs/r/schema_site_l3out.html.markdown [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    l3out_name = "example"
+    template_name = mso_site.example.template_name
+    site_id = mso_site.example.site_id
+    schema_id = mso_site.example.schema_id 
+}
+
+```
+
+## Argument Reference ##
+* `schema_id` - (Required) The schema-id where user wants to add L3out.
```
Comment (akinross) [inline website/docs/r/schema_site_l3out.html.markdown:27]: the L3Out?

File: website/docs/d/schema_site_l3out.html.markdown [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    vrf_name = mso_schema_site_vrf.example.vrf_name
+    l3out_name = mso_schema_site_l3out.example.l3out_name
+    template_name = mso_site.example.template_name
+    site_id = mso_site.example.site_id
+    schema_id = mso_site.example.schema_id
+}
+```
+
+## Argument Reference ##
+* `schema_id` - (Required) The schema-id where L3out is added.
```
Comment (akinross) [inline website/docs/d/schema_site_l3out.html.markdown:26]: the L3Out?

File: examples/schema_site_l3out/main.tf [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  vrf_name      = mso_schema_template_vrf.example.name
+}
+
+resource "mso_schema_site_l3out" "example" {
+  schema_id     = mso_schema_site.example.schema_id
+  l3out_name    = mso_schema_template_l3out.l3out.l3out_name
+  template_name = mso_schema_site.example.template_name
+  vrf_name      = mso_schema_site_vrf.example.vrf_name
+  site_id       = mso_schema_site.example.site_id
+}
```
Comment (sajagana) [inline examples/schema_site_l3out/main.tf:66]: Got "Missing required argument" and "Reference to undeclared resource" error messages. While running the example.

**Review by sajagana — CHANGES_REQUESTED** [review]

File: website/docs/d/schema_site_l3out.html.markdown [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+layout: "mso"
```
Comment (anvitha-jain) [inline website/docs/d/schema_site_l3out.html.markdown:2]: Do we need subcategory parameter here?

File: website/docs/r/schema_site_l3out.html.markdown [hunk 5]
```diff
@@ -0,0 +1,42 @@
+---
+layout: "mso"
```
Comment (anvitha-jain) [inline website/docs/r/schema_site_l3out.html.markdown:2]: Do we need subcategory parameter?

File: website/docs/r/schema_site_l3out.html.markdown [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+## Attribute Reference ##
+The only Attribute exposed for this resource is `id`. Which is set to the node name of Service Node created.
+
+## Importing ##
+
+An existing MSO Schema Site L3out can be [imported][docs-import] into this resource via its Id/path, via the following command: [docs-import]: <https://www.terraform.io/docs/import/index.html>
+
+```bash
+terraform import mso_schema_site_l3out.example {schema_id}/site/{site_id}/template/{template_name}/vrf/{vrf_name}/l3out/{l3out_name}
```
Comment (anvitha-jain) [inline website/docs/r/schema_site_l3out.html.markdown:41]: Do we need this, or should we put l3Out Dn instead?

**Pull request closed without merging**